### PR TITLE
feat(payments): use existing card for new subscriptions

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - a current subscriber should be able to use the credit card on record
   to subscribe to additional products

This commit:
 - display existing card brand and last four digit of card number
   instead of card form fields
 - update endpoint payload validation to make some new card related
   properties optional
 - update storybook to demo the UI with an existing card

## Issue that this pull request solves

Closes: #FXA-2290

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).

### CODE REVIEW

File: packages/fxa-payments-server/src/components/PaymentForm/index.tsx [hunk 1]
```diff
@@ -146,14 +172,23 @@ export const PaymentForm = ({
...
-      onSubmit={onSubmit}
-      className="payment"
-      {...{ onChange }}
-    >
+  const paymentSource = hasExistingCard ? (
+    <div className="card-details" data-testid="card-details">
+      <Localized
+        id="sub-update-card-ending"
+        vars={{
+          last: (customer as Customer).last4!,
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/PaymentForm/index.tsx:180]: `hasExistingCard` contains a check for `customer` and `last4` so this assertion should be ok.

Also, I don't know why it's `last` and not `last4` where, but I did check the ftl file and it is `last` in there.

File: packages/fxa-payments-server/src/routes/Product/SubscriptionUpgrade/index.tsx [hunk 2]
```diff
@@ -149,7 +149,7 @@ export const SubscriptionUpgrade = ({
               <Localized
                 id="payment-confirmation-cc-preview"
                 vars={{
-                  last4: cardLast4
+                  last4: cardLast4 as string,
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Product/SubscriptionUpgrade/index.tsx:152]: This is an upgrade, so the card should already exist.

File: packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx [hunk 3]
```diff
@@ -232,13 +233,38 @@ async function handleSubscriptionPayment({
...
+      }
+    );
+    return handlePaymentIntent({
+      invoiceId: createSubscriptionResult.latest_invoice.id,
+      paymentIntentStatus:
+        createSubscriptionResult.latest_invoice.payment_intent.status,
+      paymentIntentClientSecret:
+        createSubscriptionResult.latest_invoice.payment_intent.client_secret,
+      paymentMethodId:
+        createSubscriptionResult.latest_invoice.payment_intent.payment_method,
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/ProductV2/SubscriptionCreate/index.tsx:253]: Thankfully this already exists...

File: packages/fxa-payments-server/src/store/types.tsx [hunk 4]
```diff
@@ -67,15 +67,15 @@ export interface CustomerSubscription {
...
+export type Customer = {
+  billing_name?: string | null;
+  brand?: string;
+  exp_month?: string;
+  exp_year?: string;
+  last4?: string;
+  payment_type?: string;
   subscriptions: Array<CustomerSubscription>;
-}
+};
```
Comment (chenba) [inline packages/fxa-payments-server/src/store/types.tsx:78]: This matches the endpoint response validator schema.

File: packages/fxa-payments-server/src/components/PaymentForm/index.tsx (hunk 1 above)
Comment (lmorchard) [reply packages/fxa-payments-server/src/components/PaymentForm/index.tsx:180]: It's fun to be inconsistent I guess?

File: packages/fxa-payments-server/src/components/PaymentFormV2/index.tsx [hunk 5]
```diff
@@ -144,15 +158,23 @@ export const PaymentForm = ({
...
+  const paymentSource = hasExistingCard ? (
+    <div className="card-details" data-testid="card-details">
+      <Localized
+        id="sub-update-card-ending"
+        vars={{
+          last: (customer as Customer).last4!,
+        }}
+      >
+        <div
+          className={`new-sub c-card unbranded ${customer?.brand?.toLowerCase()}`}
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/components/PaymentFormV2/index.tsx:170]: Should this be something more like this? `${customer?.brand?.toLowerCase() || 'unbranded'}` 

Probably answering my own question: I think the more specific / later-defined CSS selectors win here, so it doesn't matter too much.

File: packages/fxa-payments-server/src/routes/Product/index.tsx [hunk 6]
```diff
@@ -207,6 +212,7 @@ export const Product = ({
         <SubscriptionCreate
           {...{
             profile: profile.result,
+            customer: customer.result ?? undefined,
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/routes/Product/index.tsx:215]: Oh fancy, I forgot `??` was a thing now!

**Pull request merged**